pkg/operator/ceph/nfs/config.go

```diff
@@ -48,11 +48,20 @@ func getNFSUserID(nodeID string) string {
 	return fmt.Sprintf("nfs-ganesha.%s", nodeID)
 }
 
-func getNFSClientID(n *cephv1.CephNFS, name string) string {
-	return fmt.Sprintf("client.%s", getNFSUserID(getNFSNodeID(n, name)))
+func getNFSClientID(name string) string {
+	return fmt.Sprintf("client.%s", getNFSUserID(getNFSNodeID(name)))
 }
 
-func getNFSNodeID(n *cephv1.CephNFS, name string) string {
+func getNFSNodeID(name string) string {
+	nodeid, err := k8sutil.NameToIndex(name)
+	if err != nil {
+		panic(fmt.Sprintf("failed to convert name %q to index: %v", name, err))
```

Comment:
Instead of panic how about we log an error and default to `0`? It should be a safe default